Monika After Story/game/zz_poems.rpy

```diff
@@ -103,6 +103,30 @@ init 11 python in mas_poems:
             if poem.is_seen()
         ]
 
+    def getRandomPoem(category,unseen=True):
+        """
+        Gets a random poem from the specified category
+        IN:
+            category:
+                category to search for
+
+            unseen:
+                whether or not we only want unseen poems
+                defaults to True 
+
+        OUT:
+            A random poem
+        """
+        unseen_poem_amt = len(getPoemsByCategory(category, unseen=True))
+        total_poem_amt = len(getPoemsByCategory(category, unseen=False))
+        poem_num = renpy.random.randint(0,total_poem_amt-1)
```

Comment:
this effectively rolls twice when getting an unseen poem. better to not waste a roll by changing this a little:
```python
sel_poem_len = total_poem_amt-1
if unseen:
    if unseen_poem_amt > 0:
        sel_poem_len = unseen_poem_amt-1
    else:
        unseen = False
poem_num = renpy.random.randint(0, sel_poem_len)
```